File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/command/xml/insert/InsertUrl.java
package com.kite9.k9server.command.xml.insert;

import org.w3c.dom.Element;

import com.kite9.k9server.adl.holder.pipeline.ADLDom;
import com.kite9.k9server.command.CommandException;

public class InsertUrl extends AbstractInsertCommand {

	public String uriStr;		// to insert.	

	@Override
	public Mismatch applyCommand(ADLDom in) throws CommandException {
		checkProperties();
		return doInsert(in);
	}

	@Override
	public Mismatch undoCommand(ADLDom in) throws CommandException {
		checkProperties();
		return doDelete(in);
	}
	
	@Override
	protected Element getContents(ADLDom in) {
		Element out = getForeignElementCopy(in.getDocument(), in.getUri(), uriStr, true, in);
		replaceIds(out, newId);
		return out;
	}

	@Override
	protected void checkProperties() {
		super.checkProperties();
		ensureNotNull("uriStr", uriStr);
		ensureNotNull("newId", newId);
	}
	
}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/k9server/command/xml/insert/InsertUrl.kt
package com.kite9.k9server.command.xml.insert

import com.kite9.k9server.adl.holder.pipeline.ADLDom
import com.kite9.k9server.command.Command.Mismatch
import com.kite9.k9server.command.CommandContext
import org.w3c.dom.Element

open class InsertUrl : AbstractInsertCommand() {

 	var uriStr : String? = null

    override fun applyCommand(d: ADLDom, ctx:CommandContext): Mismatch? {
        checkProperties()
        return doInsert(d, ctx)
    }

    override fun undoCommand(d: ADLDom, ctx:CommandContext): Mismatch? {
        checkProperties()
        return doDelete(d, ctx)
    }

    override fun getContents(d: ADLDom, ctx:CommandContext): Element? {
        val out = getForeignElementCopy(d.document, d.uri, uriStr!!, true, d)
        replaceIds(out, newId!!)
        return out
    }

    override fun checkProperties() {
        super.checkProperties()
        ensureNotNull("uriStr", uriStr)
        ensureNotNull("newId", newId)
    }
}